src/middlewares/check_auth.rs

```diff
@@ -117,26 +119,40 @@ where
 
         let token = token.to_string();
 
-        let components = request.app_data::<Data<AppComponents>>().unwrap().clone();
         let svc = self.service.clone();
-
         Box::pin(async move {
-            let user_id = get_user_id_from_token(components.into_inner(), &token).await;
-
-            if let Ok(user_id) = user_id {
-                {
-                    let mut extensions = request.extensions_mut();
-                    extensions.insert(user_id);
-                    extensions.insert(Token(token));
-                } // drop extension
-
-                let res = svc.call(request);
-                res.await.map(ServiceResponse::map_into_left_body)
-            } else {
-                let (request, _pl) = request.into_parts();
-                let response =
-                    HttpResponse::from_error(user_id.err().unwrap()).map_into_right_body();
-                Ok(ServiceResponse::new(request, response))
+            match request.app_data::<Data<AppComponents>>() {
+                Some(components) => {
+                    let user_id = get_user_id_from_token(
+                        components.synapse.clone(),
+                        Arc::clone(&components.users_cache),
+                        &token,
+                    )
+                    .await;
+
+                    if let Ok(user_id) = user_id {
+                        {
+                            let mut extensions = request.extensions_mut();
+                            extensions.insert(user_id);
+                            extensions.insert(Token(token));
+                        } // drop extension
+
+                        let res = svc.call(request);
+                        res.await.map(ServiceResponse::map_into_left_body)
+                    } else {
+                        let (request, _pl) = request.into_parts();
+                        let response =
+                            HttpResponse::from_error(user_id.err().unwrap()).map_into_right_body();
+                        Ok(ServiceResponse::new(request, response))
+                    }
+                }
+                None => {
+                    let (request, _pl) = request.into_parts();
+                    let response = HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR)
```

Comment:
I'd replace this:
```
let response = HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR)
.finish()
.map_into_right_body();
```
with this:
```
let response = HttpResponse::from_error(CommonError::Unknown).map_into_right_body();
```

As it's [exactly the same,](https://github.com/decentraland/social-service/blob/22ba78c99dc8a6c4851b9d317879413775633e1b/src/api/routes/v1/error.rs#L52) but it would be more consistent with the others response error we return in the block of code.